hack/make-rules/test-cmd-util.sh

```diff
@@ -221,6 +221,52 @@ setup() {
   kube::log::status "Setup complete"
 }
 
+########################################################
+# Kubectl version (--short, --client, --output) #
+########################################################
+run_kubectl_version_tests() {
+  kube::log::status "Testing kubectl version"
+
+  # create version files, one for the client, one for the server.
+  # these are the files we will use to ensure that the remainder output is correct
+  kube::test::version::object_to_file "Client" "" "client_version_test"
+  kube::test::version::object_to_file "Server" "" "server_version_test"
```

Comment:
testing the output of multiple invocations of version against itself is a little weird

in general, it would be easier to follow to do something like 
```
kubectl "${kube_flags[@]}" version ... > version.output
fgrep "expected output" version.output 
```